### PR TITLE
Lab01 todo

### PR DESCRIPTION
**Pull request by crazyloofah**

### CODE REVIEW

**Comment by brucestull** [comment]
[Lab 01: Todo](https://github.com/PdxCodeGuild/class_HB2/blob/main/3%20Django/labs/01%20Todo.md)

**Needs Work**

Comments:
* No way to add a new ToDo when there are none.
cant_create_new_todo
* Leftover residuals from 'polls' app. Please edit code to make it a ToDo app. You will either need to redo the whole thing (not a good idea) or figure out how to change the 'polls' directory and the location of 'polls' variable in the settings. It can be really challenging to fix this on existing code. Please see me if you need help. We can discuss what things need to be changed  in code if the directory name is changed. If you decide you want to change existing code from using 'polls', please do a `git` `add` and `commit` so that you can revert back to a semi-working version if something goes wrong.
* Priority 'id's hard-coded in `template` ('1', '2' , and '3') but referenced programatically in view. So application crashes. Change code so Priority 'id's in `template` are obtained programatically. This may require creating another field (which is not 'id') and use it for the 'value's in the template for Priorities. See me if you need assistance.
```
<form action="/polls/save_todo_item/" method="post">
  {% csrf_token %}
  <label for="tditem">Todo Item:</label><br />
  <input type="text" id="tditem" name="tditem" /><br />
  <input type="radio" id="high" name="priority" value="1" />
  <label for="high">High</label><br />
  <input type="radio" id="medium" name="priority" value="2" />
  <label for="meduim">Medium</label><br />
  <input type="radio" id="low" name="priority" value="3" />
  <label for="low">Low</label><br />
  <input type="submit" value="Submit" />
  <input type="reset" />
</form>
```

File: code/matt/django/lab01/todo/polls/templates/polls/todoitem_form.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+</style>
+
+<form action="/polls/save_todo_item/" method="post">
+  {% csrf_token %}
+  <label for="tditem">Todo Item:</label><br />
+  <input type="text" id="tditem" name="tditem" /><br />
+
+  <input type="radio" id="imp" name="priority" value="MOST IMPORTANT" />
+  <label for="imp">MOST IMPORTANT</label><br />
+  <input type="radio" id="high" name="priority" value="High" />
```
Comment (brucestull) [inline code/matt/django/lab01/todo/polls/templates/polls/todoitem_form.html:16]: Priority should use a drop-down for selection:
` The form for creating a todo item should also have a dropdown list for selecting the priority.`

**Review by brucestull — CHANGES_REQUESTED** [review]
# Comments:
* **[REQUIRED]** Priority choice should use a dropdown.

## **[OPTIONAL]** Other ideas which may not be required:
* The app creates a whole new Priority Item with each new ToDo. This seems to be caused be many people's desire to use `CHOICES`. `CHOICES` is not really a good use for this application. `CHOICES` would have been useful if it were included as part of the `ToDo` model, but the lab requires a `Priority` model.
    * In the documentation example about school grade levels, `CHOICES` is used to create a property of the `Student` model. And we are making a bunch of `Student`s.
    * In our lab, we only need 3 or 4 or so `Priorities` so we don't need to create new `Priority` for each `ToDo`. We can link to the original 3 or 4.

See me if you want to discuss

**Comment by crazyloofah** [comment]
Changed from radio buttons to dropdown

File: code/matt/django/lab01/todo/polls/templates/polls/todoitem_form.html (hunk 1 above)
Comment (brucestull) [reply code/matt/django/lab01/todo/polls/templates/polls/todoitem_form.html:25]: Functionality changed to drop-down \<select\>.

**Review by brucestull — APPROVED** [review]
**Completed**

**Pull request merged**